sycl/include/CL/sycl/intel/sub_group.hpp

```diff
@@ -153,11 +159,19 @@ struct sub_group {
 
   /* --- vote / ballot functions --- */
 
-  bool any(bool predicate) const {
+#if __cplusplus >= 201402L
+  [[deprecated("Use sycl::intel::any_of instead.")]]
+#endif
```

Comment:
I suggest using `__SYCL_DEPRECATED__` defined here: https://github.com/intel/llvm/blob/sycl/sycl/include/CL/sycl/detail/defines.hpp#L36.
It handles compilation with MSVC properly.

Reply:
Doesn't that definition of `__SYCL_DEPRECATED__` always expand to a message about in-order queues?

I'm guessing it should be something like this instead:
```c++
#if __cplusplus >= 201402
#define __SYCL_DEPRECATED__(message)                                           \
  [[deprecated(message)]]
#elif !defined _MSC_VER
#define __SYCL_DEPRECATED__(message)                                           \
  __attribute__((deprecated(message)))
#else
#define __SYCL_DEPRECATED__
#endif
```

Should I open a separate PR with this change?

Reply:
I think we can refactor it in the scope of this PR.

Reply:
Ok, will do.  Thanks.